core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -128,7 +129,7 @@ private void internalAddColumn(
     int parentId = TABLE_ROOT_ID;
     String fullName;
     if (parent != null) {
-      Types.NestedField parentField = schema.findField(parent);
+      Types.NestedField parentField = findField(parent);
       Preconditions.checkArgument(parentField != null, "Cannot find parent struct: %s", parent);
```

Comment:
minor note here, we may want to add in a hint about case sensitivity. `Cannot find parent struct: %s , case-sensitive: %b`

Reply:
we would have to change this all over though ...

Reply:
I think it's probably fine as set